### DISCUSSION TITLE
Improve typescript doc ( add info about ["strict":true] and ["noImplicitThis": true] )

### DISCUSSION THREAD

**Discussion by wildwind123** [vuejs/pinia#1573] [Ideas]
Hello, i think it will be helpful, if it will be on doc. I had problem with types `this.[property]` - undefined
![typescriptAAA](https://user-images.githubusercontent.com/34130873/185726482-d3d8e920-ef4e-4c4f-8457-0ff44c99e159.png)
And i solve this.
types work if  on tsconfig enabled option
`"strict":true` (this option enabled seven option (`noImplicitAny noImplicitThis alwaysStrict strictBindCallApply strictNullChecks strictFunctionTypes strictPropertyInitialization`)).
this option very bad if you migrating from js to ts etc, if this large project. And refactoring required long time.

or you can add only one option 
`"noImplicitThis": true`

I think it will be helpful for newbie user or who don't use option `"strict": true` fully.

**Comment by posva** [comment]
Sure, can you open a pr? It should go in the typescript section or in getting started

**Reply by wildwind123** [reply]
> Sure, can you open a pr? It should go in the typescript section or in getting started

Created. https://github.com/vuejs/pinia/pull/1576

But not sure, it good or not )).
